src/VisualStudio/Core/Def/Implementation/FindReferences/Entries/DocumentSpanEntry.cs

```diff
@@ -91,101 +92,82 @@ public override bool TryCreateColumnContent(string columnName, out FrameworkElem
             {
                 if (base.TryCreateColumnContent(columnName, out content))
                 {
-                    LazyToolTip.AttachTo(content, Presenter.ThreadingContext, CreateDisposableToolTip);
+                    // this lazy tooltip causes whole solution to be kept in memory until find all reference result gets cleared.
+                    // solution is never supposed to be kept alive for long time, meaning there is bunch of conditional weaktable or weak reference
+                    // keyed by solution/project/document or corresponding states. this will cause all those to be kept alive in memory as well.
+                    // probably we need to dig in to see how expensvie it is to support this
```

Comment:
This is **definitely** bad -- we've made this mistake before and caused pretty bad leaks. What's the document being used for?

Reply:
but! I am not saying solution snapshot is cheap. I am just saying it is relatively cheaper than holding everything including compilation and etc.

Reply:
LazyToolTip.AttachTo(content, Presenter.ThreadingContext, CreateDisposableToolTip);

and that func stay alive until FAR result is there.

Reply:
also, confirmed from perfview that all those solution stay alive.

Reply:
I can't find the issue, but this is not new, I raised this issue before and @sharwell had issue stating this issue and you wrote in the issue and mine long long answers why it is okay and must be.

not going to repeat those.

Reply:
> I can't find the issue, but this is not new, I raised this issue before and @sharwell had issue stating this issue and you wrote in the issue and mine long long answers why it is okay and must be.

It sounds like that may just be a bug.  I'd prefer that we just fix that.  Since you're working in this code, is there a reason we can't just do that?